drivers/volume/portworx/portworx.go

```diff
@@ -1427,6 +1443,169 @@ func getCloudSnapStatusString(status *api.CloudBackupStatus) string {
+	}
+	volumeInfos := make([]*stork_crd.VolumeInfo, 0)
+	for _, namespace := range migration.Spec.Namespaces {
+		pvcList, err := k8s.Instance().GetPersistentVolumeClaims(namespace, migration.Spec.Selectors)
+		if err != nil {
+			return nil, fmt.Errorf("error getting list of volumes to migrate: %v", err)
+		}
+		for _, pvc := range pvcList.Items {
+			if !p.isPortworxPVC(&pvc) {
+				continue
+			}
+			volumeInfo := &stork_crd.VolumeInfo{}
+			volumeInfo.PersistentVolumeClaim = pvc.Name
+			volumeInfo.Namespace = pvc.Namespace
+			volumeInfos = append(volumeInfos, volumeInfo)
+
+			volume, err := k8s.Instance().GetVolumeForPersistentVolumeClaim(&pvc)
+			if err != nil {
+				volumeInfo.Status = stork_crd.MigrationStatusFailed
+				volumeInfo.Reason = fmt.Sprintf("Error getting volume for PVC: %v", err)
+				logrus.Errorf("%v: %v", pvc.Name, volumeInfo.Reason)
+				continue
+			}
+			volumeInfo.Volume = volume
+			err = p.volDriver.CloudMigrateStart(&api.CloudMigrateStartRequest{
+				Operation: api.CloudMigrate_MigrateVolume,
+				ClusterId: clusterPair.Status.RemoteStorageID,
+				TargetId:  volume,
+			})
+			if err != nil {
+				volumeInfo.Status = stork_crd.MigrationStatusFailed
+				volumeInfo.Reason = fmt.Sprintf("Error starting migration for volume: %v", err)
+				logrus.Errorf("%v: %v", pvc.Name, volumeInfo.Reason)
+				continue
+			}
+			volumeInfo.Status = stork_crd.MigrationStatusInProgress
+			volumeInfo.Reason = fmt.Sprintf("Volume migration has started")
+		}
+	}
+
+	return volumeInfos, nil
+}
+
+func (p *portworx) GetMigrationStatus(migration *stork_crd.Migration) ([]*stork_crd.VolumeInfo, error) {
+	status, err := p.volDriver.CloudMigrateStatus()
+	if err != nil {
+		return nil, err
+	}
+
+	clusterPair, err := k8s.Instance().GetClusterPair(migration.Spec.ClusterPair)
+	if err != nil {
+		return nil, fmt.Errorf("error getting clusterpair: %v", err)
+	}
+
+	clusterInfo, ok := status.Info[clusterPair.Status.RemoteStorageID]
+	if !ok {
+		return nil, fmt.Errorf("migration status not found for remote cluster")
+	}
+
+	for _, vInfo := range migration.Status.Volumes {
+		found := false
+		for _, mInfo := range clusterInfo.List {
+			if vInfo.Volume == mInfo.LocalVolumeName {
+				found = true
+				if mInfo.Status == api.CloudMigrate_Failed {
+					vInfo.Status = stork_crd.MigrationStatusFailed
+					vInfo.Reason = fmt.Sprintf("Migration %v failed for volume", mInfo.CurrentStage)
+				} else if mInfo.CurrentStage == api.CloudMigrate_Done &&
+					mInfo.Status == api.CloudMigrate_Complete {
+					vInfo.Status = stork_crd.MigrationStatusSuccessful
+					vInfo.Reason = fmt.Sprintf("Migration succesful for volume")
+				}
+				continue
+			}
+		}
+
+		// If we didn't get the status for a volume mark it as failed
+		if !found {
+			vInfo.Status = stork_crd.MigrationStatusFailed
+		}
+	}
+
+	return migration.Status.Volumes, nil
+}
+
+func (p *portworx) CancelMigration(migration *stork_crd.Migration) error {
+	return nil
```

Comment:
Return unsupported until we added the implementation

Reply:
This is called from the controller, so can't return an error, otherwise, it'll keep retrying. Added a warning log.